### ISSUE TITLE
Can not reproduce the result

### ISSUE THREAD

**Issue by FangliangBai** [nmduc/deep-ternary#2]
Hi there,

I like your work and thanks for open the code to public.
I ran your code with the data you provided in the folder and the default parameters you set in the code but I cannot get a reconstructed image with only 50 epochs. They are very blurry. Is that because the number of training images is 2K rather than 50K?  Could you please provide more information about training?

**Comment by nmduc** [comment]
Hi, thanks for your interest.

In fact, 2K is not the number of images, but the number of patches. I don't remember but I think I only used a small number of images to generate this sample dataset. 
The training in the paper is done on a much larger dataset, with 5 millions patches extracted randomly from 50K grayscale images. The given data is just a sample, in order for others to be able to run the code, and not for reproducing the results in the paper. 

Besides, can you elaborate more on the testing images that you used, and what were the PSNRs values that you got?

**Comment by FangliangBai** [comment]
@nmduc Hi, thanks for your reply.
Yes, we are on the same pages. I thought 2K is the number of patches.
I trained the model with the following parameters, as you can see, I customised some parameters.
```
flags.DEFINE_integer("num_train_images",    30000, "total number of images")
flags.DEFINE_integer("num_val_images",      16, "total number of images for training")
flags.DEFINE_boolean("train_only",          True, "whether to train without test")
flags.DEFINE_boolean("test_only",           False, "whether to test only")
flags.DEFINE_integer("num_epoch",           12000, "train epoch num")
flags.DEFINE_integer("batch_size",          16, "batch_size")

# Learning params
flags.DEFINE_float("learning_rate",         5*1.0e-4, "initial learning rate")
flags.DEFINE_float("decay_steps",           200000, "decay_steps for learning rate")
flags.DEFINE_float("decay_rate",            0.5, "decay rate for learning rate")
flags.DEFINE_boolean("charbonnier_loss",    True, "whether to train without test")
```
However, the images I got were very blurry.
generated ![generate_12680000](https://user-images.githubusercontent.com/10924189/39052262-56f7a030-44a3-11e8-8541-697017cf2729.png)  original ![hr_10000](https://user-images.githubusercontent.com/10924189/39052263-570dda62-44a3-11e8-8c1d-6e11bf739218.png)

and the loss looks like:
![tensorboard - google chrome_001](https://user-images.githubusercontent.com/10924189/39052472-fe04fb0c-44a3-11e8-9e2a-144330a93b94.png)

I haven't use PSNR to evaluated. But so far, I don't think this reconstruction quality can have a good PSNR value.

Could you list the training setting and the hyper parameters used for training.

One more question is that I also removed batch normalisation layer and activation layer from the fully connected layers, do you think this two layers play an important role on reconstruction?

**Comment by nmduc** [comment]
@FangliangBai Hi, sorry for my late reply.
You can find all the training settings and hyper parameters in the configs/configs.py file. 

In fact, the database with 2000 patches (patches_32x32_2k.h5) is just a sample, so that people are able to run the code. To have the good results, as reported in the paper, we trained the model on a much bigger database of 5 millions patches, extracted from the ImageNet dataset. I have uploaded this big database. Please refer to the updated readme for the link and more information.

Concerning your point on the blurriness of the results: since we trained our model with the mean square error loss, we will normally have blurry results. With a better model trained using the big database, I hope you can get better results.
